### PR TITLE
Remove confusing action from List ordering example

### LINKED ISSUE #2967: List - Remove "Remove Item" from list example without checkboxes
On the DS website, we have an example under the "[Item Order](https://design-system.hpe.design/templates/lists?q=list#item-order)" example where the action toolbar button shows a list of actions : Add Item, Remove Item and Re-Order Items. It doesn't make sense to have "remove item" if we don't provide checkboxes within this example.

Some questions to ask ourselves: can list items have checkboxes?

Should we just remove the "remove item" action from this example?

Do we provide an example with a checkbox to help support the "remove item" action?

including image below:
Screen Shot 2022-10-13 at 4 09 37 PM

### Deliverables:

- [x] Dev/designer pair up to discuss whether we should include an example of list items with checkboxes OR if we should remove that item from the example.
   - Resolution (discussion between Taylor, Kenny, and Beza): Just remove the "Remove item" menu item for now. Explorations about selectable list items will be handled in a separate scope of work: https://github.com/grommet/hpe-design-system/issues/2975
- [ ] Add re-order list variant to Figma
- [ ] Update DS site

### PR DESCRIPTION
**Pull request by taysea**
[Deploy Preview](https://deploy-preview-2974--keen-mayer-a86c8b.netlify.app/templates/lists#item-order)

#### What does this PR do?

Removes "Remove item" from list ordering example since there's no way to select an item for removal. Also made a few small adjustments to the example, such as starting the example in "re-ordering" mode instead of making the user click into the menu.

Also, adjusted menu items to use sentence case to align with capitalization guidance.

#### What testing has been done on this PR?

In addition to the feature you are implementing, have you checked the following:

**General UX Checks**
- [ ] Small, medium, and large screen sizes
- [ ] Cross-browsers (FireFox, Chrome, and Safari)
- [ ] Light & dark modes
- [ ] All hyperlinks route properly

**Accessibility Checks**
- [ ] Keyboard interactions
- [ ] Screen reader experience
- [ ] Run WAVE accessibility plugin (Chrome)

**Code Quality Checks**
- [ ] Console is free of warnings and errors
- [ ] Passes E2E commit checks
- [ ] Visual snapshots are reasonable

#### What are the relevant issues?
Closes #2967

### CODE REVIEW

File: aries-site/src/examples/templates/list-views/ListOrderExample.js [hunk 1]
```diff
@@ -26,9 +26,9 @@ export const ListOrderExample = () => {
       role="application"
     >
       <Box gap="small" width="large">
-        <Text weight="bold" size="large">
+        <Heading level={2} size="small" margin="none">
```
Comment (britt6612) [inline aries-site/src/examples/templates/list-views/ListOrderExample.js:29]: where is the figma for this change? I clicked on the figma link in the first example but took me to blank list component page

Comment (taysea) [reply aries-site/src/examples/templates/list-views/ListOrderExample.js:29]: Good catch, will connect w a designer.

Comment (britt6612) [reply aries-site/src/examples/templates/list-views/ListOrderExample.js:29]: okay thank you

**Comment by britt6612** [comment]
Since were on this page should this title also be a `h2`?

Screen Shot 2022-10-17 at 5 54 33 PM

**Comment by taysea** [comment]
> Since were on this page should this title also be a `h2`?

I think this one should stay as an `h1` since it's the page title.

**Review by britt6612 — APPROVED** [review]

**Comment by KennyAtHPE** [comment]
Should we just remove the "Add Item" option?

**Comment by taysea** [comment]
> Should we just remove the "Add Item" option?

Whoops just merged, but yes will file tiny fix to just have it be one button.

**Comment by bezAdane** [comment]
I think i agree kenny  we should  maybe remove the "add item option too

**Pull request merged**